### PR TITLE
Issue 1 - Code refactoring 

### LINKED ISSUE #45: Code refactoring
According to previous call, we need to refactor code to make it in more Kotlin way.

### PR DESCRIPTION
**Pull request by vaimer**

### CODE REVIEW

File: src/main/kotlin/com/devexperts/openhack2022/cell_sandbox/game/updater/CellUpdater.kt [hunk 1]
```diff
@@ -108,10 +59,110 @@ class CellUpdater: Updater {
...
+        connection: CellConnectionState,
+        cell: CellState,
+        oldCell: CellState,
+        delta: Double
+    ) {
+        val partner = oldArea.cells[connection.partnerId]
+        if (partner != null) {
+            val partnerConnection = partner.connections[cell.id]
+            if (partnerConnection != null) {
+                // TODO why exactly 4 times?
```
Comment (david-d25) [inline src/main/kotlin/com/devexperts/openhack2022/cell_sandbox/game/updater/CellUpdater.kt]: 4 is picked as balance between physical stability and computational load

File: src/main/kotlin/com/devexperts/openhack2022/cell_sandbox/game/updater/CellUpdater.kt [hunk 2]
```diff
@@ -27,56 +23,15 @@ class CellUpdater: Updater {
...
-                            val angleOffset = stringId* PI /24 - PI /12
-                            val effectiveConnectionAngle = oldCell.angle + connection.angle + angleOffset
-                            val effectiveConnectionForceOrigin = oldCell.center + Vector2.unit(effectiveConnectionAngle) * oldCell.radius - Vector2.unit(effectiveConnectionAngle) * CELL_STICKINESS_DEPTH
-                            val effectiveConnectionForceDestination = partner.center + Vector2.unit(partner.angle + partnerConnection.angle - angleOffset) * partner.radius - Vector2.unit(partner.angle + partnerConnection.angle - angleOffset) * CELL_STICKINESS_DEPTH
-                            val connectionForceDirection = (effectiveConnectionForceOrigin to effectiveConnectionForceDestination) / 4 * delta
-                            applyImpulse(oldCell, cell, effectiveConnectionForceOrigin, connectionForceDirection)
-                        }
-                    }
-                }
+                moveCell(oldArea, connection, cell, oldCell, delta)
```
Comment (david-d25) [inline src/main/kotlin/com/devexperts/openhack2022/cell_sandbox/game/updater/CellUpdater.kt]: I think the name "moveCell" could be unclear as the function is doing only cells connection-related physics

File: src/main/kotlin/com/devexperts/openhack2022/cell_sandbox/game/renderer/CellRenderer.kt [hunk 3]
```diff
@@ -77,28 +79,36 @@ class CellRenderer: Renderer<CellState> {
...
-                    if (otherConnection != null) {
-                        val otherSurfacePoint =
-                            partner.center + Vector2.unit(partner.angle + otherConnection.angle) * partner.radius * 0.5
-                        context.strokeLine(
-                            lineStart.x, lineStart.y, otherSurfacePoint.x, otherSurfacePoint.y
-                        )
-                    }
-                }
-            }
+            addConnections(context, target, world)
```
Comment (david-d25) [inline src/main/kotlin/com/devexperts/openhack2022/cell_sandbox/game/renderer/CellRenderer.kt]: Maybe "drawConnections"? The function doesn't really "add" anything, only renders things on the screen

**Review by david-d25 — CHANGES_REQUESTED** [review]
Please see comments

File: src/main/kotlin/com/devexperts/openhack2022/cell_sandbox/game/updater/CellUpdater.kt (hunk 1 above)
Comment (vaimer) [reply src/main/kotlin/com/devexperts/openhack2022/cell_sandbox/game/updater/CellUpdater.kt]: ok got it

File: src/main/kotlin/com/devexperts/openhack2022/cell_sandbox/game/renderer/CellRenderer.kt (hunk 3 above)
Comment (vaimer) [reply src/main/kotlin/com/devexperts/openhack2022/cell_sandbox/game/renderer/CellRenderer.kt]: fixed

File: src/main/kotlin/com/devexperts/openhack2022/cell_sandbox/game/updater/CellUpdater.kt (hunk 2 above)
Comment (vaimer) [reply src/main/kotlin/com/devexperts/openhack2022/cell_sandbox/game/updater/CellUpdater.kt]: fixed

**Review by david-d25 — APPROVED** [review]

**Pull request merged**